src/jit/valuenum.cpp

```diff
@@ -1188,7 +1265,7 @@ ValueNum ValueNumStore::VNForMapSelectWork(ValueNumKind vnk, var_types typ, Valu
     VNDefFunc2Arg fstruct(VNF_MapSelect, arg0VN, arg1VN);
     if (GetVNFunc2Map()->Lookup(fstruct, &res))
     {
-        return res;
+        goto Return;
```

Comment:
I think a bunch of nested `else if` will make the code much less readable than the early returns. I don't see anything against early returns in the JIT coding conventions. Also, I think goto's can sometimes make code more readable, not less and this is such a case. If there is strong pushback to these particular goto's I'll simply revert my dumping changes and just check in the fix.